### PR TITLE
Adds phpdoc on range objectmodel

### PR DESCRIPTION
**Pull request by Matt75**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adds phpdoc on range objectmodel. Hope this can avoid mistake with usage of rangeExist() who doesn't return id of existing range. Helpful for carrier module developers too. It should return boolean but I don't want make a BC.
| Type?         | improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | Should work as before

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/13008)

### CODE REVIEW

File: classes/range/RangeWeight.php [hunk 1]
```diff
@@ -85,9 +96,11 @@ public function add($autodate = true, $null_values = false)
     }
 
     /**
-     * Get all available price ranges.
+     * Get all available price weight.
+     *
+     * @param int $id_carrier Carrier identifier
      *
-     * @return array Ranges
+     * @return array|false All range for this carrier
```
Comment (PierreRambaud) [inline classes/range/RangeWeight.php]:
```suggestion
     * @return array|bool All range for this carrier
```

or do the same for rangeExist & isOverlapping

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Little thing but :+1:  Thanks

File: classes/range/RangeWeight.php (hunk 1 above)
Comment (Matt75) [reply classes/range/RangeWeight.php]: Done

**Review by PierreRambaud — DISMISSED** [review]

File: classes/range/RangePrice.php [hunk 2]
```diff
@@ -98,6 +111,16 @@ public static function getRanges($id_carrier)
             ORDER BY `delimiter1` ASC');
     }
 
+    /**
+     * Check if a range exist for delimiter1 and delimiter2 by id_carrier or id_reference
```
Comment (eternoendless) [inline classes/range/RangePrice.php]:
```suggestion
     * Check if a range exists for delimiter1 and delimiter2 by id_carrier or id_reference
```

File: classes/range/RangePrice.php [hunk 3]
```diff
@@ -111,6 +134,16 @@ public static function rangeExist($id_carrier, $delimiter1, $delimiter2, $id_ref
             AND `delimiter1` = ' . (float) $delimiter1 . ' AND `delimiter2` = ' . (float) $delimiter2);
     }
 
+    /**
+     * Check if a range overlap another range for this carrier
```
Comment (eternoendless) [inline classes/range/RangePrice.php]:
```suggestion
     * Check if a range overlaps another range for this carrier
```

File: classes/range/RangePrice.php [hunk 4]
```diff
@@ -98,6 +111,16 @@ public static function getRanges($id_carrier)
...
 
+    /**
+     * Check if a range exist for delimiter1 and delimiter2 by id_carrier or id_reference
+     *
+     * @param int|null $id_carrier Carrier identifier
+     * @param float $delimiter1
+     * @param float $delimiter2
+     * @param int|null $id_reference Carrier reference is the initial Carrier identifier (optional)
+     *
+     * @return int|bool Number of existing range
```
Comment (eternoendless) [inline classes/range/RangePrice.php]:
```suggestion
     * @return int|false Number of matching ranges, or false on error
```

File: classes/range/RangePrice.php [hunk 5]
```diff
@@ -111,6 +134,16 @@ public static function rangeExist($id_carrier, $delimiter1, $delimiter2, $id_ref
...
 
+    /**
+     * Check if a range overlap another range for this carrier
+     *
+     * @param int $id_carrier Carrier identifier
+     * @param float $delimiter1
+     * @param float $delimiter2
+     * @param int|null $id_rang RangePrice identifier (optional)
+     *
+     * @return int|bool Number of range overlap
```
Comment (eternoendless) [inline classes/range/RangePrice.php]:
```suggestion
     * @return int|false Number of overlapping ranges, or false on error
```

File: classes/range/RangePrice.php [hunk 6]
```diff
@@ -87,7 +98,9 @@ public function add($autodate = true, $null_values = false)
     /**
      * Get all available price ranges.
      *
-     * @return array Ranges
+     * @param int $id_carrier Carrier identifier
+     *
+     * @return array|bool All range for this carrier
```
Comment (eternoendless) [inline classes/range/RangePrice.php]:
```suggestion
     * @return array|false All range for this carrier, or false on error
```

File: classes/range/RangeWeight.php [hunk 7]
```diff
@@ -85,9 +96,11 @@ public function add($autodate = true, $null_values = false)
     }
 
     /**
-     * Get all available price ranges.
+     * Get all available price weight.
```
Comment (eternoendless) [inline classes/range/RangeWeight.php]: Aren't these "Weight ranges"?

File: classes/range/RangeWeight.php [hunk 8]
```diff
@@ -85,9 +96,11 @@ public function add($autodate = true, $null_values = false)
...
+     *
+     * @param int $id_carrier Carrier identifier
      *
-     * @return array Ranges
+     * @return array|bool All range for this carrier
```
Comment (eternoendless) [inline classes/range/RangeWeight.php]:
```suggestion
     * @return array|false All ranges for this carrier, or false on error
```

File: classes/range/RangeWeight.php [hunk 9]
```diff
@@ -98,6 +111,16 @@ public static function getRanges($id_carrier)
...
 
+    /**
+     * Check if a range exist for delimiter1 and delimiter2 by id_carrier or id_reference
+     *
+     * @param int|null $id_carrier Carrier identifier
+     * @param float $delimiter1
+     * @param float $delimiter2
+     * @param int|null $id_reference Carrier reference is the initial Carrier identifier (optional)
+     *
+     * @return int|bool Number of existing range
```
Comment (eternoendless) [inline classes/range/RangeWeight.php]:
```suggestion
     * @return int|false Number of existing ranges, or false on error
```

File: classes/range/RangeWeight.php [hunk 10]
```diff
@@ -111,6 +134,16 @@ public static function rangeExist($id_carrier, $delimiter1, $delimiter2, $id_ref
             AND `delimiter1` = ' . (float) $delimiter1 . ' AND `delimiter2` = ' . (float) $delimiter2);
     }
 
+    /**
+     * Check if a range overlap another range for this carrier
```
Comment (eternoendless) [inline classes/range/RangeWeight.php]:
```suggestion
     * Check if a range overlaps another range for this carrier
```

File: classes/range/RangeWeight.php [hunk 11]
```diff
@@ -111,6 +134,16 @@ public static function rangeExist($id_carrier, $delimiter1, $delimiter2, $id_ref
...
+     *
+     * @param int $id_carrier Carrier identifier
+     * @param float $delimiter1
+     * @param float $delimiter2
+     * @param int|null $id_rang RangeWeight identifier (optional)
+     *
+     * @return int|bool Number of range overlap
```
Comment (eternoendless) [inline classes/range/RangeWeight.php]:
```suggestion
     * @return int|false Number of overlapping ranges, or false on error
```

**Review by eternoendless — CHANGES_REQUESTED** [review]

**Comment by Matt75** [comment]
@eternoendless Requested changes are done

File: classes/range/RangePrice.php [hunk 12]
```diff
@@ -87,7 +98,9 @@ public function add($autodate = true, $null_values = false)
     /**
      * Get all available price ranges.
      *
-     * @return array Ranges
+     * @param int $id_carrier Carrier identifier
+     *
+     * @return array|false All ranges for this carrier, or false on error
```
Comment (PierreRambaud) [inline classes/range/RangePrice.php:103]: but if we read `executeS` doc block
```
* @return array|false|mysqli_result|PDOStatement|resource|null
```

Comment (Matt75) [reply classes/range/RangePrice.php:103]: No opinion about that, depends of the Db driver used no ? I prefer stay simple personally

Comment (PierreRambaud) [reply classes/range/RangePrice.php:103]: I prefer too, that's why I prefer `array|bool` ^^
@eternoendless What should we do?

Comment (eternoendless) [reply classes/range/RangePrice.php:103]: By analyzing the code, in most circumstances you'll get either an array or false, so let's keep it simple.

**Review by eternoendless — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Our senseï said it all :pray: 
Thanks @Matt75

**Pull request merged**